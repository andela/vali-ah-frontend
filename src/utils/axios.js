/* eslint-disable no-param-reassign */
/**
 * An axios proxy to attach the base url and authentication token
 */
import axios from 'axios';

const axiosInstance = axios.create({
  baseURL: process.env.API_ROUTE || 'http://localhost:3000',
});

axiosInstance.interceptors.request.use((config) => {
  config.headers.Authorization = `BEARER ${localStorage.getItem('token')}`;

  return config;
});

axiosInstance.interceptors.response.use((config) => {
  config.responseBody = config?.data?.data;

  return config;
});

export default axiosInstance;
