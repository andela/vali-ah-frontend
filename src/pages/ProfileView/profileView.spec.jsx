// react libraries
import React from 'react';

import { Provider } from 'react-redux';
import axios from 'axios';

// pages
import mockStore from 'fixtures/store';
import { ViewProfile } from './index';

const store = mockStore({
  profile: {
    loading: false,
    data: {
      bio: 'Thats great',
      firstName: 'James',
      lastName: 'James',
      avatarUrl: 'MyImage',
    },
    errors: null,
  },
});

describe('Profile View', () => {
  let props;
  let wrapper;

  beforeAll(async () => {
    props = {
      profile: {
        loading: false,
        data: {
          bio: 'Thats great',
          firstName: 'James',
          lastName: 'James',
          avatarUrl: 'MyImage',
        },
        errors: null,
      },
      history: { push: jest.fn() },
      getProfileRequest: jest.fn(),
    };
    const { data } = await axios.post(
      'https://vali-1kbideas-staging.herokuapp.com/api/v1/auth/signin',
      {
        password: 'lovem!588',
        email: 'myjoy@gmail.com',
      },
    );

    const { token } = data.data;

    localStorage.setItem('token', token);
    wrapper = mount(
      <Provider store={store}>
        <ViewProfile {...props}>
          <div className="profile-container" />
        </ViewProfile>
      </Provider>,
    );
  });

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should ascertain that all the divs are existing', () => {
    expect(wrapper.find('.profile-container__header')).toHaveLength(1);
    expect(wrapper.find('.profile-container__mini-header')).toHaveLength(1);
    expect(wrapper.find('.profile-container__image-update')).toHaveLength(0);
    expect(wrapper.find('.container__image-cover')).toHaveLength(0);
    expect(wrapper.find('.profile-container__image-preview-view')).toHaveLength(
      1,
    );
    expect(wrapper.find('.profile-container__name')).toHaveLength(1);
    expect(wrapper.find('.profile-container__bio')).toHaveLength(1);
    expect(wrapper.find('.profile-container__social-detail')).toHaveLength(1);
    expect(wrapper.find('.profile-container__edit')).toHaveLength(1);
    expect(wrapper.find('.inverted orange button')).toHaveLength(0);
    expect(wrapper.find('.profile-container-link')).toHaveLength(1);
    expect(wrapper.find('.profile-container-link__published')).toHaveLength(1);
    expect(wrapper.find('.profile-container-link__bookmark')).toHaveLength(1);
    expect(wrapper.find('.profile-container-link__drafts')).toHaveLength(1);
    expect(wrapper.find('.profile-container__response')).toHaveLength(1);
  });

  it('should check if the user details are showing', () => {
    const userName = wrapper.find('#userName');
    const bio = wrapper.find('#bio');
    const profileImage = wrapper.find('#profileImg').at(0);

    expect(userName.text()).toBe(' James James');
    expect(bio.text()).toBe('Thats great');
    expect(profileImage.text()).toBe('');
  });

  it('should check if the edit is clicked', () => {
    const button = wrapper.find('.inverted');
    button.simulate('click');
  });

  it('should check if the published is clicked', () => {
    const button = wrapper.find('.profile-container-link__published');
    button.simulate('click');
  });

  it('should check if the bookmark is clicked', () => {
    const button = wrapper.find('.profile-container-link__bookmark');
    button.simulate('click');
  });

  it('should check if the drafts is clicked', () => {
    const button = wrapper.find('.profile-container-link__drafts');
    button.simulate('click');
  });
});
