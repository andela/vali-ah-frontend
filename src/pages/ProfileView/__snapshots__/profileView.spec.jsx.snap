// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile View should render properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ViewProfile
    getProfileRequest={
      [MockFunction] {
        "calls": Array [
          Array [
            "ab7055c5-f375-4ea2-a624-c98a3d0d33ff",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    profile={
      Object {
        "data": Object {
          "avatarUrl": "MyImage",
          "bio": "Thats great",
          "firstName": "James",
          "lastName": "James",
        },
        "errors": null,
        "loading": false,
      }
    }
  >
    <div>
      <div
        className="profile-container"
      >
        <div
          className="profile-container__header"
        >
          <div
            className="profile-container__mini-header"
          />
          <div
            className="profile-container__image "
          >
            <img
              alt=""
              className="profile-container__image-preview-view"
              id="profileImg"
              src="MyImage"
            />
          </div>
          <div
            className="profile-container__name"
          >
            <p
              id="userName"
            >
               
              James James
            </p>
          </div>
          <div
            className="profile-container__bio"
            id="bio"
          >
            <p>
              Thats great
            </p>
          </div>
          <div
            className="profile-container__social-detail"
          >
            <div
              className="profile-container__edit"
            >
              <button
                className="ui inverted orange button"
                onClick={[Function]}
                type="submit"
              >
                Edit Profile
              </button>
            </div>
          </div>
          <div
            className="profile-container-link"
          >
            <button
              className="profile-container-link__published  profile-active"
              onClick={[Function]}
              type="submit"
            >
              Published
            </button>
            <button
              className="profile-container-link__bookmark false"
              onClick={[Function]}
              type="submit"
            >
              Bookmarks
            </button>
            <button
              className="profile-container-link__drafts false"
              onClick={[Function]}
              type="submit"
            >
              Drafts
            </button>
          </div>
        </div>
        <div
          className="profile-container__response"
        >
          Response from Bookmarks,
        </div>
      </div>
    </div>
  </ViewProfile>
</Provider>
`;
